### PR TITLE
add wp_unslash() and sanitize_text_field() calls to inputs

### PR DESCRIPTION
**Pull request by paulschreiber**

### CODE REVIEW

**Comment by paulschreiber** [comment]
@mjangda Can we merge this?

File: co-authors-plus.php [hunk 1]
```diff
@@ -1550,7 +1550,7 @@ function wp_notify_postauthor( $comment_id, $comment_type = '' ) {
 			}
 			$notify_message .= sprintf( __( 'Spam it: %s' ), admin_url( "comment.php?action=spam&c=$comment_id" ) ) . "\r\n";
 
-			$wp_email = 'wordpress@' . preg_replace( '#^www\.#', '', strtolower( $_SERVER['SERVER_NAME'] ) );
+			$wp_email = 'wordpress@' . preg_replace( '#^www\.#', '', strtolower( sanitize_text_field( wp_unslash( $_SERVER['SERVER_NAME'] ) ) ) );
```
Comment (mjangda) [inline co-authors-plus.php]: Not necessary here.

File: php/class-coauthors-wp-list-table.php [hunk 2]
```diff
@@ -72,12 +72,12 @@ function prepare_items() {
...
 				case 'last_name':
 					$args['orderby'] = 'meta_value';
-					$args['meta_key'] = $coauthors_plus->guest_authors->get_post_meta_key( $_REQUEST['orderby'] );
+					$args['meta_key'] = $coauthors_plus->guest_authors->get_post_meta_key( sanitize_text_field( wp_unslash( $_REQUEST['orderby'] ) ) );
 					break;
 			}
 		}
-		if ( isset( $_REQUEST['order'] ) && in_array( strtoupper( $_REQUEST['order'] ), array( 'ASC', 'DESC' ) ) ) {
-			$args['order'] = strtoupper( $_REQUEST['order'] );
+		if ( isset( $_REQUEST['order'] ) && in_array( strtoupper( sanitize_text_field( wp_unslash( $_REQUEST['order'] ) ) ), array( 'ASC', 'DESC' ) ) ) {
```
Comment (mjangda) [inline php/class-coauthors-wp-list-table.php]: Not necessary here.

File: php/class-coauthors-wp-list-table.php [hunk 3]
```diff
@@ -86,7 +86,7 @@ function prepare_items() {
 				'without-linked-account'  => __( 'Without linked account', 'co-authors-plus' ),
 			);
 
-		if ( isset( $_REQUEST['filter'] ) && array_key_exists( $_REQUEST['filter'], $this->filters ) ) {
+		if ( isset( $_REQUEST['filter'] ) && array_key_exists( sanitize_text_field( wp_unslash( $_REQUEST['filter'] ) ), $this->filters ) ) {
```
Comment (mjangda) [inline php/class-coauthors-wp-list-table.php]: Not necessary here.

**Comment by mjangda** [comment]
Added some spots where the sanitize/unslash are not necessary. We don't really need them for the `wp_verify_nonce` calls either.

**Comment by paulschreiber** [comment]
* PHPCS will complain if they're not present cc @westonruter
* Do you see any downside to having them present?

**Comment by westonruter** [comment]
PHPCS should be updated to whitelist sanitization usages when an input var is used in `array_key_exists()`

**Comment by paulschreiber** [comment]
Filed https://github.com/squizlabs/PHP_CodeSniffer/issues/821

Please provide additional details there.

**Comment by paulschreiber** [comment]
Filed https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/498

Please provide additional details there.

**Comment by IanRogers-LabShift** [comment]
I would love to use this plugin, but the fact that a PR on input sanitisation has been blocked for a year is giving me pause...  Any chance some progress could be made on this please?

**Comment by paulschreiber** [comment]
@mjangda @westonruter  What's the harm in the extra wp_unslash() calls? It's safer/easier to use them everywhere than to create an exception list and manually check it on every code change.

**Comment by GaryJones** [comment]
WPCS 3.0 is nearly out, with a much-improved sniffs around this. With the conflicts in the PR, and the desire to see all of PHPCS violations fixed, I'm going to close this one out now, but it can be referred to when a fresh PR is made.

**Pull request closed without merging**